packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/watch-blocks/watch-blocks-v1-endpoint.ts

```diff
@@ -0,0 +1,329 @@
+    let listener: BlockListener;
+    let listenerType: EventType;
+
+    switch (type) {
+      case WatchBlocksListenerTypeV1.Full:
+        listener = async (blockEvent) => this.monitorFullCallback(blockEvent);
+        listenerType = "full";
+        break;
+      case WatchBlocksListenerTypeV1.Filtered:
+        listener = async (blockEvent) =>
+          this.monitorFilteredCallback(blockEvent);
+        listenerType = "filtered";
+        break;
+      case WatchBlocksListenerTypeV1.Private:
+        listener = async (blockEvent) =>
+          this.monitorPrivateCallback(blockEvent);
+        listenerType = "private";
+        break;
+      case WatchBlocksListenerTypeV1.CactusTransactions:
+        listener = async (blockEvent) =>
+          this.monitorCactusTransactionsCallback(blockEvent);
+        listenerType = "full";
+        break;
+      default:
+        // Will not compile if any type was not handled by above switch.
+        const unknownType: never = type;
+        throw new Error(
+          `Unknown block listen type - '${unknownType}'. Check name and connector version.`,
+        );
+    }
+
+    if (!listener || !listenerType) {
+      // Should never happen
+      throw new Error("Could not determine listener or listenerType.");
+    }
+
+    return { listener, listenerType };
+  }
+
+  /**
+   * Subscribe to new blocks on fabric ledger, push them to the client via socketio.
+   *
+   * @param options Block monitoring options.
+   */
+  public async subscribe(options: WatchBlocksOptionsV1) {
+    const { socket, log } = this;
+    const clientId = socket.id;
+    log.info(`${WatchBlocksV1.Subscribe} => clientId: ${clientId}`);
+    log.debug(
+      "WatchBlocksV1.Subscribe args: channelName:",
+      options.channelName,
+      ", startBlock:",
+      options.startBlock,
+      ", type: ",
+      options.type,
+    );
+
+    try {
+      Checks.truthy(options.channelName, "Missing channel name");
+      const network = await this.config.gateway.getNetwork(options.channelName);
+
+      const { listener, listenerType } = this.getBlockListener(options.type);
+
+      log.debug("Subscribing to new blocks... listenerType:", listenerType);
+      // @todo Add support for checkpointer (long-term improvement)
+      // https://hyperledger.github.io/fabric-sdk-node/release-2.2/module-fabric-network.Checkpointer.html
+      await network.addBlockListener(listener, {
+        startBlock: options.startBlock,
+        type: listenerType,
+      });
+
+      socket.on("disconnect", async (reason: string) => {
+        log.info(
+          "WebSocket:disconnect => reason=%o clientId=%s",
+          reason,
+          clientId,
+        );
+        network.removeBlockListener(listener);
+        this.close();
+      });
+
+      socket.on(WatchBlocksV1.Unsubscribe, () => {
+        log.info(`${WatchBlocksV1.Unsubscribe} => clientId: ${clientId}`);
+        this.close();
+      });
+    } catch (error: any) {
```

Comment:
@outSH Could this work with the error being typed as `unknown`?

Reply:
@petermetz Sure

Reply:
@outSH Cheers!